Automattic/wp-calypso
# Reader: data layer / reducer for feed searching

V1 support for searching feeds.

It contains:
 - data layer w/ http middleware for querying feeds
 - basic reducer for adding search results to the state tree
 - query component/selector

Things missing and to add in future prs:
 - pagination support for a query results (backend change as well)
 - purging results or utilizing an lru cache for holding on to query results? 
 - I've left out data transformation because we don't currently really transform feeds on the calypso side anywhere and don't want to make decisions there just yet

To Test:
1. run unit tests
2. manually dispatch a search using `dispatch( { type: 'READER_FEED_SEARCH_REQUEST', payload: { query: 'chicken' } } )`.  verify it shows up in the state tree under `state.reader.feedSearches`
